Backend.md

```diff
@@ -0,0 +1,167 @@
+
+# Phantasm Renderer Backend
+
+PRB is an abstraction over Vulkan and D3D12, aiming to be succinct without restricting access to the underlying APIs within reason. Primary goals:
+
+- Small API surface
+- High performance
+- Explicit lifetimes
+- Heavy multithreading
+- Low boilerplate
+
+## Objects
+
+The core of PRB is the backend. This is the only type in the library with methods<sup>[1](#footnote1)</sup>. Everything happening with regards to the real graphics API is instructed through the backend. The two backends  (`BackendD3D12` and `BackendVulkan`) share a virtual interface and can be used interchangably.
+
+PRB has four main objects, accessed using lightweight, POD handles.
+
+1. `handle::resource`
+
+    A texture, render target, or buffer of any form
+
+2. `handle::pipeline_state`
+
+    Shaders, their input shape, and various pipeline configuration options form a pipeline_state. Graphics, compute, or raytracing.
+
+3. `handle::shader_view`
+
+    Shader views are a combination of resource views, for access in a shader.
+
+4. `handle::command_list`
+
+    Represents a recorded command list ready for GPU submission.
+
+### Commands
+
+Commands do the heavy lifting in communicating with a PRB backend. They are simple structs, which are written contiguously into a piece of memory ("software command buffer"). This memory is then passed to the backend,
+
+```C++
+handle::command_list recordCommandList(std::byte* buffer, size_t size);
```

Comment:
could this be a span?

Reply:
It wouldn't make sense to use a span as it is not semantically an array of `std::byte`, but an opaque buffer. Neither the callsite nor the internals would benefit from range checks, and the backends would immediately extract the pointer and the size anyways.